src/main/kotlin/com/pubnub/api/endpoints/pubsub/Subscribe.kt

```diff
@@ -54,6 +55,8 @@ class Subscribe internal constructor(pubnub: PubNub) : Endpoint<SubscribeEnvelop
             queryParams["state"] = pubnub.mapper.toJson(it)
         }
 
+        PubNubUtil.maybeAddEeQueryParam(pubnub.configuration, queryParams)
```

Comment:
What about extracting all code that set query params in this method to separate method having name e.g. addQueryParams ?

Reply:
done